src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -929,7 +960,9 @@ private void visitClassOrInterface(String name, FunctionType ftype) {
         emit("implements");
         emitCommaSeparatedInterfaces(it);
       }
-      visitObjectType(ftype, ftype.getPrototype());
+
+      boolean implementsIArrayLike = any(ftype.getOwnImplementedInterfaces(), IARRAYLIKE_PREDICATE);
```

Comment:
Only the own implemented interfaces (immediate implemented?), or also transitively?